driver/src/driver/stablex_driver.rs

```diff
@@ -39,14 +39,21 @@ impl<'a> StableXDriver<'a> {
             solution
         };
 
-        let submitted = if solution.executed_sell_amounts.iter().sum::<u128>() > 0 {
-            self.contract.submit_solution(batch, orders, solution)?;
-            info!("Successfully applied solution to batch {}", batch);
-            true
+        let current_objective_value = self.contract.get_current_objective_value()?;
+        let submitted = if let Some(surplus) = solution.surplus {
+            if current_objective_value < surplus {
+                self.contract.submit_solution(batch, orders, solution)?;
+                info!("Successfully applied solution to batch {}", batch);
+                true
+            } else {
+                info!("Not submitting trivial solution for batch {}", batch);
+                false
+            }
         } else {
             info!("Not submitting trivial solution for batch {}", batch);
             false
         };
```

Comment:
This looks like duplicate code. Was this intended?

Reply:
Yes, this is what I was referring to here:
```
I was searching here for a more elegant solution, but I can not find it.

Thread discussing it:
rust-lang/rfcs#929 (comment)
```